xskillscore/tests/test_deterministic.py

```diff
@@ -122,28 +128,64 @@ def test_correlation_metrics_xr(a, b, dim, weight_bool, weights, metrics):
 
     axis = _a.dims.index(new_dim)
     if metric in temporal_only_metrics:
-        res = _metric(_a.values, _b.values, axis, skipna=False)
+        res = _metric(_a.values, _b.values, axis, skipna=skipna)
     else:
-        res = _metric(_a.values, _b.values, _weights, axis, skipna=False)
+        res = _metric(_a.values, _b.values, _weights, axis, skipna=skipna)
     expected = actual.copy()
     expected.values = res
     assert_allclose(actual, expected)
 
 
+@pytest.mark.parametrize("metrics", correlation_metrics)
+@pytest.mark.parametrize("dim", AXES)
+@pytest.mark.parametrize("weight_bool", [True, False])
+@pytest.mark.parametrize("skipna", [True, False])
+@pytest.mark.parametrize("has_nan", [True, False])
+def test_correlation_metrics_daskda_same_npda(
+    a_dask, b_dask, dim, weight_bool, weights_dask, metrics, skipna, has_nan
+):
+    """Test whether correlation metric for xarray functions can be lazy when
+    chunked by using dask and give same results as DataArray with numpy array."""
+    a = a_dask.copy()
+    b = b_dask.copy()
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "a, b",
    [
        (
            pytest.lazy_fixture("a_dask"),
            pytest.lazy_fixture("b_dask"),
        ),
        (
            pytest.lazy_fixture("a"),
            pytest.lazy_fixture("b"),
        ),
        (
            pytest.lazy_fixture("a_nan"),
            pytest.lazy_fixture("b_nan"),
        ),
    ],
)
@pytest.mark.parametrize("metrics", correlation_metrics)
@pytest.mark.parametrize("dim", AXES)
@pytest.mark.parametrize("weight_bool", [True, False])
@pytest.mark.parametrize("skipna", [True, False])
def test_correlation_metrics_daskda_same_npda(
    a, b, dim, weight_bool, weights_dask, metrics, skipna, has_nan
):
    """Test whether correlation metric for xarray functions can be lazy when
    chunked by using dask and give same results as DataArray with numpy array."""
    a = a.copy() # maybe delete
    b = b.copy()
```

Reply:
requires: `pip install pytest-lazy-fixture`

maybe also rename `a` to `a_1d` so we can use `a` in the parametrized tests as input. cannot give `a2` and `a` in this example both the same name.

Reply:
this example now with `a_nan` is probably not better than `has_nan` [True, False]. `has_nan` has the advantage that the small function changes the code. on the other hand you could also add `a_nan_dask` to the list of inputs, but this would be growing and growing...

there will be no 100% clean and everything matching solution in this...